go-controller/observability-lib/sampledecoder/sample_decoder.go

```diff
@@ -133,74 +135,60 @@ func (d *SampleDecoder) DecodeCookieIDs(obsDomainID, obsPointID uint32) (string,
-		return "", fmt.Errorf("failed to get message for db object %v", dbObj)
+	var event any
+	switch o := dbObj.(type) {
+	case *nbdb.ACL:
+		event = newACLEvent(o)
+	}
+	if event == nil {
+		return nil, fmt.Errorf("failed to build network event for db object %v", dbObj)
 	}
-	return msg, nil
+	return event, nil
 }
 
-func getMessage(dbObj interface{}) string {
-	switch o := dbObj.(type) {
-	case *nbdb.ACL:
-		var action string
-		switch o.Action {
-		case nbdb.ACLActionAllow, nbdb.ACLActionAllowRelated, nbdb.ACLActionAllowStateless:
-			action = "Allowed"
-		case nbdb.ACLActionDrop:
-			action = "Dropped"
-		case nbdb.ACLActionPass:
-			action = "Delegated to network policy"
-		default:
-			action = "Action " + o.Action
-		}
-		actor := o.ExternalIDs[libovsdbops.OwnerTypeKey.String()]
-		var msg string
-		switch actor {
-		case libovsdbops.AdminNetworkPolicyOwnerType:
-			msg = fmt.Sprintf("admin network policy %s, direction %s", o.ExternalIDs[libovsdbops.ObjectNameKey.String()], o.ExternalIDs[libovsdbops.PolicyDirectionKey.String()])
-		case libovsdbops.BaselineAdminNetworkPolicyOwnerType:
-			msg = fmt.Sprintf("baseline admin network policy %s, direction %s", o.ExternalIDs[libovsdbops.ObjectNameKey.String()], o.ExternalIDs[libovsdbops.PolicyDirectionKey.String()])
-		case libovsdbops.MulticastNamespaceOwnerType:
-			msg = fmt.Sprintf("multicast in namespace %s, direction %s", o.ExternalIDs[libovsdbops.ObjectNameKey.String()], o.ExternalIDs[libovsdbops.PolicyDirectionKey.String()])
-		case libovsdbops.MulticastClusterOwnerType:
-			msg = fmt.Sprintf("cluster multicast policy, direction %s", o.ExternalIDs[libovsdbops.PolicyDirectionKey.String()])
-		case libovsdbops.NetpolNodeOwnerType:
-			msg = "default allow from local node policy, direction ingress"
-		case libovsdbops.NetworkPolicyOwnerType:
-			msg = fmt.Sprintf("network policy %s, direction %s", o.ExternalIDs[libovsdbops.ObjectNameKey.String()], o.ExternalIDs[libovsdbops.PolicyDirectionKey.String()])
-		case libovsdbops.NetpolNamespaceOwnerType:
-			msg = fmt.Sprintf("network policies isolation in namespace %s, direction %s", o.ExternalIDs[libovsdbops.ObjectNameKey.String()], o.ExternalIDs[libovsdbops.PolicyDirectionKey.String()])
-		case libovsdbops.EgressFirewallOwnerType:
-			msg = fmt.Sprintf("egress firewall in namespace %s", o.ExternalIDs[libovsdbops.ObjectNameKey.String()])
-		case libovsdbops.UDNIsolationOwnerType:
-			msg = fmt.Sprintf("UDN isolation of type %s", o.ExternalIDs[libovsdbops.ObjectNameKey.String()])
-		}
-		return fmt.Sprintf("%s by %s", action, msg)
-	default:
-		return ""
+func newACLEvent(o *nbdb.ACL) *model.ACLEvent {
+	actor := o.ExternalIDs[libovsdbops.OwnerTypeKey.String()]
+	event := model.ACLEvent{
+		Action: o.Action,
+		Actor:  actor,
+	}
+	switch actor {
+	case libovsdbops.AdminNetworkPolicyOwnerType, libovsdbops.BaselineAdminNetworkPolicyOwnerType, libovsdbops.NetworkPolicyOwnerType:
+		event.Name = o.ExternalIDs[libovsdbops.ObjectNameKey.String()]
```

Comment:
this part for network policy is confusing, because it has both namespace and name

Reply:
do you mean, AdminNetworkPolicyOwnerType / BaselineAdminNetworkPolicyOwnerType / NetworkPolicyOwnerType ?
Previously each of them only mentioned name and direction in their associated message, that's why I did not mention a namespace. You're saying there should be a namespace there too ?

Reply:
better like that: https://github.com/ovn-kubernetes/ovn-kubernetes/pull/4818/commits/bf1db3464aaca129bc926d88dc1cd876689a3315 ?

Reply:
it is called "name" in the externalIDs, but for netpol it contains namespace+name. Since externalIDs don't distinguish namespace vs name, it kinda makes sense, but if your struct has separate namespace and name, then it probably should be parsed